rs/nns/sns-wasm/src/sns_wasm.rs

```diff
@@ -324,36 +325,30 @@ where
         &self,
         #[allow(unused_variables)] get_wasm_metadata_payload: GetWasmMetadataRequestPb,
     ) -> GetWasmMetadataResponsePb {
-        // TODO[NNS1-2997]: Enable this feature on mainnet.
-        #[cfg(feature = "test")]
         let result = self.get_wasm_metadata_impl(get_wasm_metadata_payload);
-
-        #[cfg(not(feature = "test"))]
-        let result = Err("get_wasm_metadata is not implemented yet.".to_string());
-
         GetWasmMetadataResponsePb::from(result)
     }
 
     pub fn get_wasm_metadata_impl(
         &self,
         get_wasm_metadata_payload: GetWasmMetadataRequestPb,
     ) -> Result<Vec<MetadataSection>, String> {
-        let module_hash = <[u8; 32]>::try_from(get_wasm_metadata_payload)?;
+        let hash = <[u8; 32]>::try_from(get_wasm_metadata_payload)?;
+
+        let Some(wasm) = self.read_wasm(&hash) else {
+            return Err(format!("Cannot find WASM stored under key `{:?}`.", hash));
+        };
 
-        let wasm_metadata = self.read_wasm_metadata_or_err(&module_hash)?;
+        let wasm_metadata = Self::read_wasm_metadata_or_err(&wasm)?;
```

Comment:
Okay, so we are populating the metadata in the index a post_upgrade hook, but we're still reading it here directly from the wasm?

Reply:
Oops. Thanks for catching this.

Reply:
Done, PT another L @max-dfinity